### PR TITLE
Feature - Menu and Menu.Item component created

### PR DESCRIPTION
**Pull request by MoiyadKayda**
## Description
I have tried to create these component. Please verify if this is the correct way. I have also made some changes in the Button component added `ref` and `stopPropagation (in Link as well)`.
**Still need to figure out why portal doesn't reopen with correct position.**

## Screenshots:
image

## Checklist:

- [ ] I have updated the CHANGELOG with a summary of my changes
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Comment by MoiyadKayda** [comment]
There are some type error because of the change in Button component.
I was not able to figure out how to get rid of them.

File: src/organisms/menu/component.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import { getClassName } from '@kibalabs/core';
+import { IMultiChildProps, OptionalProppedElement } from '@kibalabs/core-react';
+import styled from 'styled-components';
+
+import { IListItemTheme, ListItem } from '../../atoms';
```
Comment (krishan711) [inline src/organisms/menu/component.tsx:7]: why is this an organism and not a molecule? Also the menuitem seems like it should really be a particle. this should really looke just like list and listitem i think?

Comment (krishan711) [reply src/organisms/menu/component.tsx:7]: hm in general i think the challenge here is its very hard for me to tell if this is on the right track or not. i'd suggest not continuing on code and going back to the google doc - pls try document your thinking in terms of theming and particle/molecule/organism there.

Comment (krishan711) [reply src/organisms/menu/component.tsx:7]: As always, should not be made in the PR, they should be made before but here there are loads of "surprises" i wasnt expecting.

File: src/atoms/button/component.tsx [hunk 2]
```diff
@@ -78,12 +78,13 @@ export interface IButtonProps extends IComponentProps<IButtonTheme> {
   onClicked?(): void;
 }
 
-export const Button = (props: IButtonProps): React.ReactElement => {
-  const onClicked = (): void => {
+export const Button = React.forwardRef((props: IButtonProps, ref: React.ForwardedRef<HTMLButtonElement>): React.ReactElement => {
```
Comment (krishan711) [inline src/atoms/button/component.tsx:81]: why is this needed?

File: src/molecules/menuItem/component.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+import { ISingleAnyChildProps } from '@kibalabs/core-react';
+
+import { defaultMoleculeProps, IMoleculeProps } from '../moleculeProps';
+
+interface IMenuItemTheme {
+}
+
+export interface IMenuItemProps extends IMoleculeProps<IMenuItemTheme>, ISingleAnyChildProps {
```
Comment (krishan711) [inline src/molecules/menuItem/component.tsx:10]: this file looks very wrong to me

File: src/atoms/button/component.tsx (hunk 2 above)
Comment (MoiyadKayda) [reply src/atoms/button/component.tsx:81]: So that we can attach the portal to a Button

File: src/molecules/menuItem/component.tsx (hunk 3 above)
Comment (MoiyadKayda) [reply src/molecules/menuItem/component.tsx:10]: I tried to put the list.item inside here instead of returning `null` but it was not working here.

File: src/organisms/menu/component.tsx (hunk 1 above)
Comment (MoiyadKayda) [reply src/organisms/menu/component.tsx:7]: I will update the doc file with whatever I thought about the menu.

File: src/atoms/button/component.tsx (hunk 2 above)
Comment (krishan711) [reply src/atoms/button/component.tsx:81]: we should only ever attach to a box - otherwise everything will need a ref at some point

Comment (MoiyadKayda) [reply src/atoms/button/component.tsx:81]: Yeah, you are right
The way to do this is to pass the prop from Button to box, right?

Comment (krishan711) [reply src/atoms/button/component.tsx:81]: yep if by "pass" you mean wrap the button in a box wherever you need it then put the ref on the box

Comment (MoiyadKayda) [reply src/atoms/button/component.tsx:81]: No, if I wrap a box on a button then how can someone click on the button.
I mean that we take the props in button and then pass it to the box component.

Comment (MoiyadKayda) [reply src/atoms/button/component.tsx:81]: That way we don't need a forward ref on button

Comment (krishan711) [reply src/atoms/button/component.tsx:81]: that doesnt make sense - if a button is in a box it can still be clicked.

Comment (krishan711) [reply src/atoms/button/component.tsx:81]: pls try and show me if it fails

Comment (MoiyadKayda) [reply src/atoms/button/component.tsx:81]: Ooh yeah, I thought of something else. If we click on a box it should propagate the click event to the parent also.
Simply a transparent box can wrap a button for this to work.

Comment (krishan711) [reply src/atoms/button/component.tsx:81]: yes exactly - box are just divs. Just FYI on this as well the box doesnt need to propogate the click explicitly - it will happen without interference from us.
